iso15118/secc/states/iso15118_2_states.py

```diff
@@ -1021,6 +1021,19 @@ async def process_message(
 
         authorization_req: AuthorizationReq = msg.body.authorization_req
 
+        # [V2G2-475] The message 'AuthorizationRes' shall contain the ResponseCode
+        # 'FAILED_ChallengeInvalid' if the challenge response contained in the
+        # AuthorizationReq message in attribute GenChallenge is not valid versus
+        # the provided GenChallenge in PaymentDetailsRes.
+        if authorization_req.gen_challenge:
```

Comment:
if the field gen_challenge is addressable but can be None, then do we need this if clause?
cant we just have the one after?

Reply:
i see, gen_challenge is optional, if we remove this statement, we can face an issue when ev doesnt send gen_challenge value. if ev doesnt send it, the value will be None, and automatically we return as FAILED_CHALLENGE_INVALID, that is not a valid case. Am i wrong?

Reply:
you are not checking if the gen_challenge attribute exists in the authorization_req data structure. Instead, you are already assuming that the attribute exists but that its value can be None or not: this may be ok, depends how the pydantic model constructs the data. Is it the case that if the gen_challenge is None we will still be able to access the `gen_challenge`attribute? if so, then we dont need that first if clause, because in the one after we do the check as well

Reply:
There is one use case being missed because of the if condition above - the case where GenChallenge is not send at all in AuthorizationReq in PnC.

I believe you added the check as the field is optional in EIM. It would continue to work in EIM if the check is removed anyway because of the reason @Andre mentioned above - but might throw an error in case a GenChallenge is accidentally sent in EIM AuthorizationReq. It may be better to ignore the field check in EIM completely?

```
if self.comm_session.selected_auth_option is AuthEnum.PNC_V2:
```

Reply:
@shalinnijel2 should not we terminate the session with FAILED_CHALLENGE_INVALID in this case as well. I think, we might not need this check...

Reply:
Yes, true  - letting that pass through would also be a violation of the spec - in which case simply removing the if condition as discussed above would suffice.